### PR TITLE
[RM Ch09] Add CI/CD design requirements

### PR DESCRIPTION
**Pull request by collivier**
It adds a couple of CI/CD design requirements to allow smoothly deploy
and test the cloud infrastructures and the hosted network functions.

Signed-off-by: Cédric Ollivier <cedric.ollivier@orange.com>

### CODE REVIEW

File: doc/ref_model/chapters/chapter09.md [hunk 1]
```diff
@@ -234,6 +234,43 @@ auto.cicd.014 | The CD pipeline must be able to automatically promote software f
...
+[Gitlab CI/CD](https://docs.gitlab.com/ee/ci/). But they are many other
+ways where deployment and test jobs can constraint the end users from the
+build servers to the artifact management. Any manual operation is discouraged
+whatever it's about the deployment or the test resources.
+
+The following requirements also aims at deploying smoothly and easily all CI/CD
+toolchains via simple playbooks as targeted by the Reference Conformance
+suites currently leveraging [XtestingCI](https://galaxy.ansible.com/collivier/xtesting).
+
+Ref #           | Description                                                                                                    |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter09.md:258]:
```suggestion
Ref #           | Description                                                                                                    | Comments |
```

File: doc/ref_model/chapters/chapter09.md [hunk 2]
```diff
@@ -234,6 +234,43 @@ auto.cicd.014 | The CD pipeline must be able to automatically promote software f
...
+----------------|----------------------------------------------------------------------------------------------------------------|
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter09.md:259]:
```suggestion
----------------|----------------------------------------------------------------------------------------------------------------|---|
```

File: doc/ref_model/chapters/chapter09.md [hunk 3]
```diff
@@ -234,6 +234,43 @@ auto.cicd.014 | The CD pipeline must be able to automatically promote software f
...
+design.cicd.001 | The pipeline must allow chaining all deployment and test operations from baremetal to the network services     |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter09.md:260]:
```suggestion
design.cicd.001 | The pipeline must allow chaining of independent CI/CD jobs | For example, all deployment and test operations from baremetal to Kubernetes, OpenStack, to the network services   |
```

File: doc/ref_model/chapters/chapter09.md [hunk 4]
```diff
@@ -234,6 +234,43 @@ auto.cicd.014 | The CD pipeline must be able to automatically promote software f
...
+design.cicd.002 | The pipeline should be modular to only execute a subset of the overall steps                                   |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter09.md:261]:
```suggestion
design.cicd.002 | The pipeline should be modular | this allows execution of jobs independently of others, for example, start with an existing OpenStack  deployment   |
```

File: doc/ref_model/chapters/chapter09.md [hunk 5]
```diff
@@ -234,6 +234,43 @@ auto.cicd.014 | The CD pipeline must be able to automatically promote software f
...
+design.cicd.003 | The pipeline must decouple the deployment and the test steps                                                   |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter09.md:262]:
```suggestion
design.cicd.003 | The pipeline must decouple the deployment and the test steps                                                   |  |
```

File: doc/ref_model/chapters/chapter09.md [hunk 6]
```diff
@@ -234,6 +234,43 @@ auto.cicd.014 | The CD pipeline must be able to automatically promote software f
...
+design.cicd.004 | The pipeline should leverage the jobs artifacts proposed by the CI/CD tool                                     |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter09.md:263]:
```suggestion
design.cicd.004 | The pipeline should leverage the job artifacts specified by the CI/CD tool                                     |  |
```

File: doc/ref_model/chapters/chapter09.md [hunk 7]
```diff
@@ -234,6 +234,43 @@ auto.cicd.014 | The CD pipeline must be able to automatically promote software f
...
+design.cicd.005 | The pipeline must execute all relevant Reference Conformance suites as they are                                |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter09.md:264]:
```suggestion
design.cicd.005 | The pipeline must execute all relevant Reference Conformance suites without modification                                |  |
```

File: doc/ref_model/chapters/chapter09.md [hunk 8]
```diff
@@ -234,6 +234,43 @@ auto.cicd.014 | The CD pipeline must be able to automatically promote software f
...
+design.cicd.006 | All single jobs must not ask for a specific CI/CD tool                                                         |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter09.md:265]:
```suggestion
design.cicd.006 | Software vendors/providers must utilise operator provided CI/CD tools                                                        |  |
```

File: doc/ref_model/chapters/chapter09.md [hunk 9]
```diff
@@ -234,6 +234,43 @@ auto.cicd.014 | The CD pipeline must be able to automatically promote software f
...
+design.cicd.007 | All single jobs must be packaged as containers                                                                 |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter09.md:266]:
```suggestion
design.cicd.007 | All jobs must be packaged as containers                                                                 |   |
```

File: doc/ref_model/chapters/chapter09.md [hunk 10]
```diff
@@ -234,6 +234,43 @@ auto.cicd.014 | The CD pipeline must be able to automatically promote software f
...
+design.cicd.008 | All jobs should leverage a common execution to allow templating all deployment and test steps                  |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter09.md:267]:
```suggestion
design.cicd.008 | All jobs must leverage a common execution to allow templating all deployment and test steps                  |  |
```

File: doc/ref_model/chapters/chapter09.md [hunk 11]
```diff
@@ -234,6 +234,43 @@ auto.cicd.014 | The CD pipeline must be able to automatically promote software f
...
+design.cicd.009 | The deployment jobs must publish all outputs as artifacts in the best formats (OpenStack RC, kubeconfig, etc.) |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter09.md:268]:
```suggestion
design.cicd.009 | The deployment jobs must publish all outputs as artifacts in a specified format | For example, OpenStack RC, kubeconfig, yaml, etc. Anuket shall specify formats in RC |
```

File: doc/ref_model/chapters/chapter09.md [hunk 12]
```diff
@@ -234,6 +234,43 @@ auto.cicd.014 | The CD pipeline must be able to automatically promote software f
...
+design.cicd.010 | All test jobs must download all inputs as artifacts in the best formats (OpenStack RC, kubeconfig, etc.)       |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter09.md:269]:
```suggestion
design.cicd.010 | Test jobs must provide all inputs as artifacts in a specified format | For example, OpenStack RC, kubeconfig, yaml, etc. Anuket shall specify formats in RC       |
```

File: doc/ref_model/chapters/chapter09.md [hunk 13]
```diff
@@ -234,6 +234,43 @@ auto.cicd.014 | The CD pipeline must be able to automatically promote software f
...
+design.cicd.011 | All test jobs should conform with the Reference Conformance test case integration requirements                 |
```
Comment (pgoyal01) [inline doc/ref_model/chapters/chapter09.md:270]:
```suggestion
design.cicd.011 | Test jobs must conform with the Reference Conformance test case integration requirements                 |  |
```

File: doc/ref_model/chapters/chapter09.md [hunk 14]
```diff
@@ -234,6 +234,43 @@ auto.cicd.014 | The CD pipeline must be able to automatically promote software f
 
 <p align="center"><b>Table 9-4:</b> Automation CI/CD</p>
 
+
+#### 9.5.2.2 CI/CD Design Requirements
```
Comment (karinesevilla) [inline doc/ref_model/chapters/chapter09.md:238]: @collivier 
section numbering ->9.5.2.3

**Review by pgoyal01 — APPROVED** [review]

**Review by TFredberg — APPROVED** [review]

**Review by karinesevilla — APPROVED** [review]

**Review by walterkozlowski — APPROVED** [review]

**Pull request merged**